### PR TITLE
Add read write access check

### PR DESCRIPTION
**Pull request by nightfury1204**
#10 #8 
- Add read and write access check
- Log error for KeyStoreNotFount

### CODE REVIEW

File: pkg/worker/options.go [hunk 1]
```diff
@@ -33,9 +33,12 @@ type WorkerOptions struct {
 	// ca cert file for vault api client, if vault used a self signed certificate
 	CaCertFile string
 
-	// If InSecureTLS true, then it will skip tls verification when communicating with vault server
+	// If InSecureTLS is true, then it will skip tls verification when communicating with vault server
 	InSecureTLS bool
 
+	// If SkipReadWriteAccessCheck is true, then it will skip read/write access check before initializing the vault
+	SkipReadWriteAccessCheck bool
```
Comment (tamalsaha) [inline pkg/worker/options.go]: Do we need this flag?

File: pkg/vault/vault.go [hunk 2]
```diff
@@ -153,6 +159,31 @@ func (u *vault) Init() error {
...
+	glog.Infoln("Testing the read/write access...")
+	key := "mock-test-vault-unsealer"
+	data := []byte("test")
+
+	err := u.keyStore.Set(key, data)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write data for key(%s)", key)
+	}
+
+	_, err = u.keyStore.Get(key)
```
Comment (tamalsaha) [inline pkg/vault/vault.go]: Should you not delete here so that test data does not stay around ?

Comment (nightfury1204) [reply pkg/vault/vault.go]: I think you are right. Let me work on that.

File: pkg/worker/options.go (hunk 1 above)
Comment (nightfury1204) [reply pkg/worker/options.go]: I keep this flag, if user don't want to give delete permission to unsealer

**Pull request merged**